lib/api/Healthcheck.js

```diff
@@ -101,6 +115,174 @@ class Healthcheck {
             }
         });
     }
+
+    /**
+     * Get deep healthcheck
+     * A deep healthcheck will send entries to BackbeatProducer to produce to
+     * Kafka, catch these entries using the same global Kafka topic used for
+     * Backbeat CRR, but using a new BackbeatConsumer for each health request
+     * @param {function} cb - callback(error, data)
+     * @return {undefined}
+     */
+    getDeepHealthcheck(cb) {
+        if (this._probe !== null) {
+            // TODO: Based on previous request, given how deep healthcheck
+            //  currently works, cannot handle parallel requests above cluster
+            //  count. Will need to refactor this
+            const error = {
+                method: 'Healthcheck.getDeepHealthcheck',
+                error: 'Another request still processing',
+            };
+            return cb(error);
+        }
+        if (!this._crrProducer || !this._crrStatusProducer) {
+            const error = {
+                method: 'Healthcheck.getDeepHealthcheck',
+                error: 'Backbeat Producer not set',
+            };
+            return cb(error);
+        }
+
+        const start = Date.now();
+        const id = `deephealthcheck-${uuid().replace(/-/g, '')}`;
+        this._probe = new Probe(id);
+
+        return async.waterfall([
+            done => this._getTopicPartitions(done),
+            (partitions, done) => {
+                this._probe.setupStore(partitions);
+                this._createEntries(id, partitions, done);
+            },
+            (entries, done) => this._crrProducer.send(entries, done),
+            done => this._createConsumer('deephealthcheck-group', done),
+        ], (err, consumer) => {
+            if (err || consumer === undefined) {
+                const error = {
+                    method: 'Healthcheck.getDeepHealthcheck',
+                    error: err,
```

Comment:
May be good to add a custom error if `consumer` is undefined. Perhaps something alluding to the fact that the BackbeatConsumer was not created somehow. Then again, I think this case may ever happen because it is an instantiated class.